### PR TITLE
Add poap tables and data fetching api

### PR DESCRIPTION
**Pull request by crabsinger**
### <samp>🤖 Generated by Copilot at 6ff63c0</samp>

This pull request adds the functionality to fetch and sync POAP data from the external API to the Hasura database. It creates three new tables: `poap_events`, `poap_holders`, and `external_data_fetches` to store the POAP data and the fetching status. It also adds a new file `api-lib/poap/poap-api.ts` that contains the logic for fetching and syncing the data using the `POAP_API_KEY` environment variable. It updates the `.env.example` file to include a comment about the variable.

### <samp>🤖 Generated by Copilot at 6ff63c0</samp>

> _`POAP_API_KEY`_
> _fetch and sync data with Hasura_
> _autumn of events_

### <samp>🤖 Generated by Copilot at 6ff63c0</samp>

*  Add logic for fetching and syncing POAP data from external API to Hasura database ([link](https://github.com/coordinape/coordinape/pull/2384/files?diff=unified&w=0#diff-1c94cb78d152557eb07f7b63c6a870a6358304fd2f3d0255335c2781f7804761R1-R212))
* Create tables for storing POAP data and external data fetches in Hasura database ([link](https://github.com/coordinape/coordinape/pull/2384/files?diff=unified&w=0#diff-e78e54b8e555ca375d04d3b0ea41530e2206eeb4c15aa7e9ac75a470938de2b3R1-R3), [link](https://github.com/coordinape/coordinape/pull/2384/files?diff=unified&w=0#diff-7ac72d6974707cb6d954b321eb3eb41f5ab477530aaeaf2205a58b10dbca25a2R1-R3), [link](https://github.com/coordinape/coordinape/pull/2384/files?diff=unified&w=0#diff-b190ec6f099af9d360f4d8d7e9434a22f490284e28fdc44caa55196d1a4b0ed9R1-R3), [link](https://github.com/coordinape/coordinape/pull/2384/files?diff=unified&w=0#diff-9d80a3e604cac3d26b1245c88c25b904a2508e2357a3c4ed3f48c8118e0f5d32R1), [link](https://github.com/coordinape/coordinape/pull/2384/files?diff=unified&w=0#diff-8eee18ff668617e3e970ebbb46c2bfbb8287898cad49a2f696af8456e43e82c1R1-R32), [link](https://github.com/coordinape/coordinape/pull/2384/files?diff=unified&w=0#diff-6e420cb76f82af383311a6067e520c5422f9ddf7d633e6c95b8262ed419d5a27R1), [link](https://github.com/coordinape/coordinape/pull/2384/files?diff=unified&w=0#diff-f3638acffaee065e1d0bef4d343349e4ea5cefd11981702091e21286ee43d89aR1-R22), [link](https://github.com/coordinape/coordinape/pull/2384/files?diff=unified&w=0#diff-309adcb54b1826656026b882c6de9d3c408e83ac8516e5152cc9f2511e0e8b91R1), [link](https://github.com/coordinape/coordinape/pull/2384/files?diff=unified&w=0#diff-1e5cd8ee646b4ed9bb4561dd9bb2d118ea3216c39b7f066e813af3fe4edcc52fR1-R15))
* Add comment to `.env.example` file to indicate need for `POAP_API_KEY` variable ([link](https://github.com/coordinape/coordinape/pull/2384/files?diff=unified&w=0#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cR123))

### CODE REVIEW

File: hasura/migrations/default/1698099787726_create_table_public_external_data_fetches/up.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+CREATE TABLE "public"."external_data_fetches" ("id" bigserial NOT NULL,
```
Comment (crabsinger) [inline hasura/migrations/default/1698099787726_create_table_public_external_data_fetches/up.sql:1]: s//address_data_fetches

remove id, use address as pk

File: api-lib/poap/poap-api.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+      operationName: 'insert_external_data_fetches_poap',
+    }
+  );
+
+  console.log('inserted some poap data', {
+    insert_poap_holders,
+    insert_poap_events,
+    insert_external_data_fetches_one,
+  });
+  return {
```
Comment (crabsinger) [inline api-lib/poap/poap-api.ts:207]: remove

File: hasura/metadata/databases/default/tables/public_external_data_fetches.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+table:
+  name: external_data_fetches
```
Comment (CryptoGraffe) [inline hasura/metadata/databases/default/tables/public_external_data_fetches.yaml:2]: this table is gone

Comment (crabsinger) [reply hasura/metadata/databases/default/tables/public_external_data_fetches.yaml]: good catch. fixed.

**Pull request merged**